src/SonarQube.Client/Services/SonarQubeService.cs

```diff
@@ -59,56 +60,61 @@ public bool HasOrganizationsFeature
             }
         }
 
-        public bool IsConnected => this.connection != null;
+        public bool IsConnected => this.sonarqubeClient != null;
 
         public async Task ConnectAsync(ConnectionInformation connection, CancellationToken token)
         {
-            if (this.connection != null)
+            if (this.sonarqubeClient != null)
             {
                 throw new InvalidOperationException("This operation expects the service not to be connected.");
             }
 
-            var ConnectionRequest = new ConnectionRequest
+            var connectionRequest = new ConnectionRequest
             {
                 Authentication = connection.Authentication,
                 ServerUri = connection.ServerUri,
                 Login = connection.UserName,
                 Password = connection.Password
             };
 
-            var credentialsValidationResult = await this.sonarqubeClient.ValidateCredentialsAsync(ConnectionRequest, token);
+            var client = this.sonarqubeClientFactory.Create(connectionRequest);
+
+            var credentialsValidationResult = await client.ValidateCredentialsAsync(token);
             credentialsValidationResult.EnsureSuccess();
 
             if (!credentialsValidationResult.Value.IsValid)
             {
                 throw new Exception("Invalid credentials."); // TODO: Provide better exception
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') 'System.Exception' should not be thrown by user code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS112)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this 'TODO' comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1135)